interfaces/mount/change.go

```diff
@@ -0,0 +1,118 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package mount
+
+import (
+	"path"
+	"sort"
+	"strings"
+)
+
+// Action represents a mount action (mount, remount, unmount, etc).
+type Action string
+
+const (
+	Mount   Action = "mount"
+	Unmount Action = "umount"
+	// Remount when needed
+)
+
+// Change describes a change to the mount table (action and the entry to act on).
+type Change struct {
+	Entry  Entry
+	Action Action
+}
+
+// NeededChanges computes the changes required to change current to desired mount entries.
+//
+// The current and desired profiles is a fstab like list of mount entries. The
+// lists are processed and a "diff" of mount changes is produced. The mount
+// changes, when applied in order, transform the current profile into the
+// desired profile.
+func NeededChanges(current, desired []Entry) []Change {
+	var changes []Change
+	// Copy both as we will want to mutate them.
+	c := make([]Entry, len(current))
```

Comment:
I know this go appreciation for short name but I find c and d here a bit too much,  cur and des ?